### ISSUE TITLE
Why aren't torch.functional.sigmoid and torch.nn.functional.relu deprecated like torch.nn.functional.tanh?

### ISSUE THREAD

**Issue by blednotik** [pytorch/pytorch#22117]
In the `torch.nn.functional`, you deprecated `tanh` but not `sigmoid` or `relu`.  Why is this?  Is it an inconsistency to be fixed, or there are important considerations supporting it?  Is it okay to replace `torch.nn.functional.relu` by `torch.relu` (and same about `sigmoid`)  in my code?

PS: at current code, `torch.nn.sigmoid` is deprecated too, but not `relu`.

**Comment by fmassa** [comment]
`tanh` is a mathematical function used outside of neural networks, and goes hand in hand with `cosh` and `sinh`. That's why it should live in the `torch` namespace.

For `relu` or `sigmoid`, their scope is much more limited, and thus we want them in the `nn` namespace.

The fact that `relu` exists in the `torch` namespace is an artifact. We should only keep the one present in the `nn` namespace

**Comment by blednotik** [comment]
Actually, in the current code, `sigmoid` is also deprecated in `nn` [https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1390](url).  But not `relu`.

**Comment by fmassa** [comment]
`relu` in `torch.relu` is the one that should be removed

**Comment by blednotik** [comment]
What about the mathematical function "ramp function"?

**Comment by VitalyFedyunin** [comment]
Need to decide in group and write final decision.

**Comment by VitalyFedyunin** [comment]
We will follow up with the depreciation plans doc.

**Comment by fmassa** [comment]
@blednotik note that we already have `torch.clamp`, and that `x.clamp(min=0)` is equivalent to `.relu()`

**Comment by mruberry** [comment]
Closing this due to age and lack of a clear issue.
